s/cpprim.ss

```diff
@@ -5408,9 +5408,46 @@
                     ,(build-fixnum->flonum e-x values)
                     (if ,(%type-check mask-flonum type-flonum ,e-x)
                         ,e-x
-                        ,(build-libcall #t src sexpr real->flonum e-x `(quote real->flonum))))))]))
-    (define-inline 3 $real->flonum
-      [(x who) (build-$real->flonum src sexpr x who)])
+                        ,(build-libcall #t src sexpr $real->flonum `(quote real->flonum) e-x)))))])
+      (define-inline 3 $real->flonum
+        [(who x) (build-$real->flonum src sexpr who x)])
+      (define-inline 2 inexact
+        [(e-x)
+         (if (known-flonum-result? e-x)
+             e-x
+             (bind #t (e-x)
+               `(if ,(%type-check mask-fixnum type-fixnum ,e-x)
+                    ,(build-fixnum->flonum e-x values)
+                    (if ,(build-simple-or
+                          (%type-check mask-flonum type-flonum ,e-x)
+                          (%typed-object-check mask-inexactnum type-inexactnum ,e-x))
+                        ,e-x
+                        ,(build-libcall #t src sexpr inexact e-x)))))])
+      (define-inline 2 exact->inexact
```

Comment:
This seems like too much code duplication. Add a `build-exact->inexact` parameterized over the library entry to call?